src/org/opendatakit/briefcase/ui/reused/AggregateServerConnectionDialog.java

```diff
@@ -0,0 +1,68 @@
+package org.opendatakit.briefcase.ui.reused;
+
+import java.awt.Cursor;
+
+import org.opendatakit.briefcase.model.ServerConnectionInfo;
+import org.opendatakit.briefcase.model.TerminationFuture;
+import org.opendatakit.briefcase.ui.MessageStrings;
+import org.opendatakit.briefcase.ui.ODKOptionPane;
+import org.opendatakit.briefcase.util.ServerConnectionTest;
+
+public class AggregateServerConnectionDialog {
+  final AggregateServerConnectionDialogForm form;
+  private boolean asTarget;
+  private boolean isSuccessful;
+  private ServerConnectionInfo serverInfo = null;
+  private TerminationFuture terminationFuture = new TerminationFuture();
+
+  AggregateServerConnectionDialog(AggregateServerConnectionDialogForm aggregateServerConnectionDialogForm, boolean asTarget) {
+    form = aggregateServerConnectionDialogForm;
+    this.asTarget = asTarget;
+
+    form.addConnectButtonCallback(e -> verifyParams());
+  }
+
+  private void verifyParams() {
+    final ServerConnectionInfo info = new ServerConnectionInfo((form.getURLFieldText()).trim(),
+        form.getUsernameFieldText(), form.getPasswordFieldPassword());
+
+    form.disableConnectButton();
+    form.disableCancelButton();
+    String errorString;
+
+    Cursor saved = form.getCursor();
+    try {
+      form.setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
+      form.paint(form.getGraphics());
+      terminationFuture.reset();
+      ServerConnectionTest backgroundAction = new ServerConnectionTest(info, terminationFuture, asTarget);
```

Comment:
We need to develop our own ServerConnectionTest service. You can start by duplicating it. The reason behind it is that we want to get rid of the side-effects of the original implementation. While you do that, get rid of the `TerminationFuture` and all the code that uses it.

Reply:
`TerminationFuture` is used in a lot of places and most of the util ops are based on it. Should I just delete it?